### PR TITLE
Add timestamp to singleton

### PR DESCRIPTION
**Pull request by almogdepaz**
add confirmation timestamp to singleton

### CODE REVIEW

**Review by Quexington — DISMISSED** [review]
lgtm

File: chia/data_layer/dl_wallet_store.py [hunk 1]
```diff
@@ -38,7 +52,8 @@ async def create(cls: Type[_T_DataLayerStore], db_wrapper: DBWrapper) -> _T_Data
                 " confirmed tinyint,"
                 " confirmed_at_height int,"
                 " proof blob,"
-                " generation int)"  # This first singleton will be 0, then 1, and so on.  This is handled by the DB.
+                " generation int,"
+                " timestamp int)"  # This first singleton will be 0, then 1, and so on.  This is handled by the DB.
```
Comment (altendky) [inline chia/data_layer/dl_wallet_store.py:56]:
```suggestion
                " generation int,"  # This first singleton will be 0, then 1, and so on.  This is handled by the DB.
                " timestamp int)"
```

File: chia/data_layer/dl_wallet_store.py [hunk 2]
```diff
@@ -14,6 +14,20 @@
 _T_DataLayerStore = TypeVar("_T_DataLayerStore", bound="DataLayerStore")
 
 
+def _row_to_singleton_record(row: Row) -> SingletonRecord:
```
Comment (altendky) [inline chia/data_layer/dl_wallet_store.py:17]: `SingletonRecord.from_row()`?

**Review by altendky — DISMISSED** [review]
restamping quex's approval after a minor comment change

File: chia/data_layer/data_layer.py [hunk 3]
```diff
@@ -137,12 +137,12 @@ async def get_ancestors(self, node_hash: bytes32, store_id: bytes32) -> List[Int
...
 
-    async def get_root(self, store_id: bytes32) -> Tuple[Optional[bytes32], Status]:
+    async def get_root(self, store_id: bytes32) -> Optional[SingletonRecord]:
         latest = await self.wallet_rpc.dl_latest_singleton(store_id, True)
         if latest is None:
             self.log.error(f"Failed to get root for {store_id.hex()}")
-            return None, Status.PENDING
-        return latest.root, Status.COMMITTED
+            return None
+        return latest
```
Comment (Nikolaj-K) [inline chia/data_layer/data_layer.py:145]: The `return None` is redundant now.

**Review by altendky — DISMISSED** [review]

**Review by altendky — APPROVED** [review]

**Pull request merged**